### PR TITLE
Scaling

### PR DESCRIPTION
**Pull request by tim-dinsdale**
Tolerance on snapping, some docs

### CODE REVIEW

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community! We require contributors to sign a [Contributor License Agreement](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/75530375/Legal+Requirements#LegalRequirements-ContributorLicenseAgreement) and we don't seem to have CLAs on file for these contributors to the Pull Request: (@tim-dinsdale). In order for your PR to be reviewed and merged, please follow the directions at the link above.

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If there are any questions, please don't hesitate to [get in touch with our Infrastructure Support team](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/77955298/Getting+in+Touch#GettinginTouch-InfrastructureSupport). /CC @finos-admin

File: src/provider/model/DesktopSnapGroup.ts [hunk 1]
```diff
@@ -241,8 +248,9 @@ export class DesktopSnapGroup {
                 // If a window is not visible it cannot be adjacent to anything. This also allows us
                 // to avoid the questionable position tracking for hidden windows.
                 return false;
-            } else if (distance.border(0) && Math.abs(distance.maxAbs) > MIN_OVERLAP) {
-                // The overlap check ensures that only valid snap configurations are counted
+            } else if (distance.border(1) && Math.abs(distance.maxAbs) > MIN_OVERLAP) {
```
Comment (tomrobertsOF) [inline src/provider/model/DesktopSnapGroup.ts]: Minor, but is it worth defining this as a constant like we do with some other things. Maybe `ADJACENCY_DISTANCE` or something similar?

File: src/provider/model/DesktopSnapGroup.ts [hunk 2]
```diff
@@ -231,6 +231,13 @@ export class DesktopSnapGroup {
             }
         }
 
+        /**
+         * Are the two DesktopEntitys adjacent? True if they are in the same tabgroup, they
+         * are both visible, or they are close enough to be within tolerance.
+         * @param win1 one window
```
Comment (tomrobertsOF) [inline src/provider/model/DesktopSnapGroup.ts]: We should be careful about the use of the word "window" here. `DesktopEntity` is not strictly limited to windows.

File: src/provider/model/DesktopSnapGroup.ts [hunk 3]
```diff
@@ -231,6 +231,13 @@ export class DesktopSnapGroup {
             }
         }
 
+        /**
+         * Are the two DesktopEntitys adjacent? True if they are in the same tabgroup, they
+         * are both visible, or they are close enough to be within tolerance.
```
Comment (tomrobertsOF) [inline src/provider/model/DesktopSnapGroup.ts]: I think this is not exactly what the checks below result to. The actual check is "true if they are windows in the same tab group, false if they are not both visible,  true if they are both visible and within the fuzz distance. Probably a more succinct way to put it, but we should make sure it is logically correct above readability.

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community! We require contributors to sign a [Contributor License Agreement](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/75530375/Legal+Requirements#LegalRequirements-ContributorLicenseAgreement) and we don't seem to have CLAs on file for these contributors to the Pull Request: (@tim-dinsdale). In order for your PR to be reviewed and merged, please follow the directions at the link above.

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If there are any questions, please don't hesitate to [get in touch with our Infrastructure Support team](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/77955298/Getting+in+Touch#GettinginTouch-InfrastructureSupport). /CC @finos-admin

File: src/provider/model/DesktopSnapGroup.ts (hunk 1 above)
Comment (tim-dinsdale) [reply src/provider/model/DesktopSnapGroup.ts]: fair, done

File: src/provider/model/DesktopSnapGroup.ts (hunk 2 above)
Comment (tim-dinsdale) [reply src/provider/model/DesktopSnapGroup.ts]: fair, done

File: src/provider/model/DesktopSnapGroup.ts (hunk 3 above)
Comment (tim-dinsdale) [reply src/provider/model/DesktopSnapGroup.ts]: yeah, I agree, done

File: src/provider/snapanddock/Constants.ts [hunk 4]
```diff
@@ -17,6 +17,11 @@ export const ANCHOR_DISTANCE = 100;
  */
 export const MIN_OVERLAP = 50;
 
+/**
+ * To calculate window adjacency, we use a tiny fuzz distance to deal with sub-pixel window drift
+ */
+export const ADJACENCY_FUZZ_DISTANCE = 1;
```
Comment (bryangaleOF) [inline src/provider/snapanddock/Constants.ts]: I've spotted snapping failing at 150% unless this is bumped up to 2. Good to bump up?

Comment (tim-dinsdale) [reply src/provider/snapanddock/Constants.ts]: 👍

**Pull request merged**